### ISSUE TITLE
Frame separation fails when an error frame occurs before a normal frame.  

### ISSUE THREAD

**Issue by Guswib** [Sensirion/arduino-core#8]
Hi, I have a MCF SFC6xxxx device using RS485

When I sending the command following command to the device:
 0x7E, 0x0, 0x0, 0x1, 0x1, 0xFD, 0x7E
It sends back:
**0x7E**, 0xFE, 0xFF, 0xF9, 0xF9, 0xFD, **0x7E**, **0x7E**, 0x0, 0x0, 0x0, 0x4, 0x0, 0x0, 0x0, 0x0, 0xFB, **0x7E**
which are two frames following each other. The first frame seems to be a error response, while the second frame seems to be the normal response.
The function "SensirionShdlcCommunication::receiveFrame" fails to split the two frames and I only get a timeout. If I read 7 bytes from the RX-stream before executing the SensirionShdlcCommunication::receiveFrame, I get a expected response. 
So, the issue is the initial error frame. It might be due to that the 4th byte in the frame, i.e the "length", is not 0, which should be the case for a error frame. So, the reading of incoming bytes passes over the end-of-frame byte instead of breaking.

**Comment by sdmueller** [comment]
Hi @Guswib

The command you are sending looks correct to me.

Do you get similar behavior when sending other commands?
And what happens if you use the SFx6xxx specific library here: https://github.com/Sensirion/arduino-uart-sfx6xxx
It uses this library.

**Comment by Guswib** [comment]
Yes, all commands result in the initial error frame. Discarding this initial frame, all values are reasonable. 

Thanks for the link to the library. I tried it out and I get the following response(see below).
I also added the "getSetpoint", and some more printouts. As you can see, I get a lot of error messages, but not always. For example, I can see that I can set the setPoint. So, it seems that the device send this error frame for some strange reason.

12:15:50.031 -> readAveragedMeasuredValue:----------------------- 
12:15:50.031 -> Error trying to execute readAveragedMeasuredValue(): Wrong command in return frame
12:15:51.041 -> SetPoint:----------------------- 
12:15:51.041 -> Error trying to execute getSetpoint(): Checksum is wrong
12:15:51.082 -> readAveragedMeasuredValue:----------------------- 
12:15:51.082 -> averagedMeasuredValue: -0.00
12:15:52.061 -> SetPoint:----------------------- 
12:15:52.061 -> Error trying to execute getSetpoint(): Not enough space in buffer
12:15:52.104 -> readAveragedMeasuredValue:----------------------- 
12:15:52.104 -> Error trying to execute readAveragedMeasuredValue(): Wrong command in return frame
12:15:53.112 -> SetPoint:----------------------- 
12:15:53.112 -> Error trying to execute getSetpoint(): Checksum is wrong
12:15:53.112 -> readAveragedMeasuredValue:----------------------- 
12:15:53.112 -> averagedMeasuredValue: -0.00
12:15:54.110 -> SetPoint:----------------------- 
12:15:54.110 -> Error trying to execute getSetpoint(): Not enough space in buffer
12:15:54.110 -> readAveragedMeasuredValue:-----------------------

**Comment by Guswib** [comment]
I tested another device. The same issues persist....

**Comment by sdmueller** [comment]
Hm, this is most probably an issue with the connection where some frames or parts of it are getting lost and not the sensor itself.

What is your exact setup and what Arduino are you using?

**Comment by Guswib** [comment]
I have recreated the exact setup that is shown in the example, while using a arduino Mega. 
p.s. I have created a pull request to fix the issue with the error frame.

**Comment by LeonieFierz** [comment]
Thank you for your pull request to fix the error frame, which is now merged to the master.

Sorry for not coming back earlier. Does your problem with the SFx6xxx still persists? We found that unfortunately the RS485 to TTL Converter U094 from M5Stack is not very stable, so this could also be the source of the errors.
